### ISSUE TITLE
feat: Decouple signing

### ISSUE THREAD

**Issue by udf2457** [rdimitrov/go-tuf-metadata#62]
### Is your feature request related to a problem? Please describe.

From my brief reading of the `metadata` code, at present your library appears to make the same mistake as `theupdateframework/go-tuf` in that key signing is tighly coupled to primitive and crude on-disk keyfiles.

This means that real-world secure key storage such as PKCS#11 (https://github.com/theupdateframework/go-tuf/issues/427), AWS KMS (https://github.com/theupdateframework/go-tuf/issues/525) and others e.g. Yubikey are not readily supported and require hacky work-around kludges to work (e.g. manually hacking json files).

### Describe the solution you'd like

Of course support for signing from local keyfiles stored on disk should remain, but integration with real world applications where the private key is stored in a non-exfilterable format should be supported.

**Comment by MDr164** [comment]
Sounds like a good idea, we should create a signer package that simply takes in bytes and outputs signed bytes instead of signing only files.

**Comment by rdimitrov** [comment]
@udf2457 - It is definitely something we should have in go-tuf/go-tuf-metadata 👍 I'm supportive of refactoring the existing signer mechanism to incorporate such support. Let me know if I can help you or someone else who is interested in contributing with that. Currently we are finalising the testing (@MDr164 and @ivanayov helped a lot) and eventually we'll start copying the codebase to go-tuf where they'll live in parallel for a start so it's less of a change. I can try to squeeze something else in my agenda but unfortunately I cannot promise anything, at least for next few weeks.

**Comment by rdimitrov** [comment]
Closing in favour of https://github.com/theupdateframework/go-tuf/issues/594
